include/IECoreMaya/LiveScene.h

```diff
@@ -185,7 +186,8 @@ class LiveScene : public IECore::SceneInterface
 		// Register callbacks for custom attributes.
 		// The names function will be called during attributeNames and hasAttribute.
 		// The readAttr method is called if the names method returns the expected attribute, so it should return a valid Object pointer or raise an Exception.
-		static void registerCustomAttributes( NamesFn namesFn, ReadAttrFn readFn );
+		// If the mightHave function is specified, it will be called before names function for early out, to see if the names function can return the expected attribute.
+		static void registerCustomAttributes( NamesFn namesFn, ReadAttrFn readFn, MightHaveFn mightHaveFn = 0 );
```

Comment:
This breaks binary compatibility - to maintain compatibility you would need to add an overload with the extra argument. You could then reimplement the original method to just call the overload with a NULL third argument.